### DISCUSSION TITLE
How to create a Deno module from a public GitHub repository?

### DISCUSSION THREAD

**Discussion by wmwsagara** [denoland/deno#14950] [General]
Hi

How to create a private Deno module from a public GitHub repository? I want to import the module from file system. This is mainly for security reasons.

Public GitHub repo: https://github.com/solana-labs/solana-web3.js

Appreciate your reply.

Best regards
Sagara

**Comment by hironichu** [comment]
You are probably looking to use this https://deno.land/manual/linking_to_external_code/private

**Reply by wmwsagara** [reply]
Thanks hironichu. But this is not what I want.

That page says "There maybe instances where you want to load a remote module that is located in a private repository ...".

I don't have the "remote module" in first place. 

My question is how to make the module in first place out of a "public" repository which does not require any permission to access?

Best regards
Sagara

**Reply by hironichu** [reply]
Hm not quite sure if i understand it well but,

You can download the source code of that remote module, once available locally you can import it from your filesystem, 
`import {something} from "../path/to/local/module.(ts, js)`

**Reply by wmwsagara** [reply]
Indeed, what I'm trying is not trivial stuff. It is best you try on your side try to import.
